src/PandoraMonitoring.cc

```diff
@@ -991,6 +992,109 @@ void PandoraMonitoring::ViewEvent()
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+bool CheckPathExists(const std::string& path)
+{
+    struct stat info;
+
+    if (stat(path.c_str(), &info) != 0)
+        return false;
+
+    return (info.st_mode & S_IFDIR) != 0;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool MakePathIfNeeded(const std::string& path)
+{
+    // We use mkdir here, not stat to avoid race conditions. If the folder
+    // exist, we will just fail and pick that up in the following switch.
+    mode_t mode = 0755;
+    int ret = mkdir(path.c_str(), mode);
+
+    if (ret == 0)
+        return true;
+
+    // Check the error code for making the folder.
+    // There is three cases we care about:
+    //   - The parent doesn't exist, so lets go back one and try and make that
+    //     folder first by recursing.
+    //   - The folder already exists, so we don't need to do anything.
+    //   - Any other error we should just stop. The full list can be find in
+    //     the mkdir man page. This includes stuff like too long a path, disk
+    //     quotas, permission issues and more.
+    switch (errno) {
+        case ENOENT:
+            // Parent didn't exist, try to create it
+            {
+                int pos = path.find_last_of('/');
+                if (pos == std::string::npos)
+                    return false;
+                if (!MakePathIfNeeded( path.substr(0, pos) ))
+                    return false;
+            }
+            // Now, try to create again
+            return 0 == mkdir(path.c_str(), mode);
```

Comment:
Prefer parentheses around the Boolean expression that will be returned. e.g. `return (0 == mkdir...);`